embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/tasks/ndk/CollectSharedObjectFilesTask.kt

```diff
@@ -0,0 +1,103 @@
+package io.embrace.android.gradle.plugin.tasks.ndk
+
+import io.embrace.android.gradle.plugin.config.ProjectType
+import io.embrace.android.gradle.plugin.config.UnitySymbolsDir
+import io.embrace.android.gradle.plugin.tasks.EmbraceTask
+import io.embrace.android.gradle.plugin.tasks.EmbraceTaskImpl
+import io.embrace.android.gradle.plugin.tasks.il2cpp.UnitySymbolFilesManager
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFiles
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * Collects .so files from the project, depending on the project type.
+ */
+abstract class CollectSharedObjectFilesTask @Inject constructor(
```

Comment:
```suggestion
abstract class CopyFilesTask @Inject constructor(
```

Would suggest making this abstract by supplying the location for the files to copy as a property of the task. We could also put this in the `common` package rather than `ndk`

Reply:
good idea!